### PR TITLE
rewrite registration login logout.

### PR DESCRIPTION
**Pull request by atreidas1**

### CODE REVIEW

File: app/components/groups/group-storage.service.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
 ;(function(){
   angular
     .module("todo")
```
Comment (alekseyC64) [inline app/components/groups/group-storage.service.js]: shouldn't it be on one line with `angular` word? like `angular.module("todo")`

File: app/components/logining/login.controller.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    .controller("LoginController", LoginController);
+
+  LoginController.$inject = ["SessionService"];
+
+  function LoginController(SessionService) {
+    var self = this;
+    self.signIn = signIn;
+    self.isMsgHide = true;
+    self.loginError = loginError;
+    self.user = {};
```
Comment (ghost) [inline app/components/logining/login.controller.js]: It's not right to initialize objects with empty object. If it's no data - `null` is the best option

File: app/components/registration/register.controller.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    .controller("RegisterController", RegisterController);
+
+  RegisterController.$inject = ["SessionService"];
+
+  function RegisterController(SessionService) {
+    var self = this;
+    self.signUp = signUp;
+    self.isMsgHide = true;
+    self.registerError = registerError;
+    self.user = {};
```
Comment (ghost) [inline app/components/registration/register.controller.js]: the same

File: app/components/session.service.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    self.getUser();
+
+    function getUser() {
+      $http.get(API.BASE + API.AUTH + API.INFO)
+        .then(function (response) {
+          self.user.data = response.data.user;
+          $state.go("groups.list");
+        })
+        .catch(function () {});
```
Comment (ghost) [inline app/components/session.service.js]: there is no sense in such `catch` code.

File: app/components/session.service.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        .then(function () {
+          callCallback(successFn);
+          self.user.data = undefined;
+          $state.go("login");
+        })
+        .catch(function () {})
+    }
+  }
+
+  function callCallback(fn, args) {
```
Comment (ghost) [inline app/components/session.service.js]: it's look like kind of hack, why do we need this? You already use promises, it's enough

File: app/components/groups/group.contoller.js [hunk 6]
```diff
@@ -4,9 +4,9 @@
 
     .controller("GroupController", GroupController)
 
-    GroupController.$inject = ["GroupStorage", "$scope"]
+    GroupController.$inject = ["GroupStorage", "$scope", "SessionService"]
 
-    function GroupController(GroupStorage, $scope){
+    function GroupController(GroupStorage, $scope, SessionService){
       $scope.groupView = "list"
```
Comment (NikitaSedyx) [inline app/components/groups/group.contoller.js]: it's should be equal `SessionService.user.userView` for me

File: app/components/registration/register.controller.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+  function RegisterController(SessionService) {
+    var self = this;
+    self.signUp = signUp;
+    self.isMsgHide = true;
+    self.registerError = registerError;
+    self.user = null;
+
+    function signUp() {
+      var res = SessionService.registration(self.user).then(registerError);
```
Comment (NikitaSedyx) [inline app/components/registration/register.controller.js]: mb catch for errors?

File: app/components/logining/login.jade [hunk 8]
```diff
@@ -0,0 +1,17 @@
+.row
+  .col-md-6.col-md-offset-3
+    .alert.alert-danger.errMsg(ng-hide="loginCtrl.isMsgHide") {{loginCtrl.allertMsg}}
```
Comment (NikitaSedyx) [inline app/components/logining/login.jade]: use dashes in class and id names, pls

File: app/components/logining/login.jade [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+.row
+  .col-md-4.col-md-offset-4
+    .panel.panel-primary.login-form
+      .panel-heading Sign In:
+      form.panel-body(ng-submit="loginCtrl.signIn()")
+        .form-group
+          label(for="login") Logi:
```
Comment (NikitaSedyx) [inline app/components/logining/login.jade]: does it work without id in input filed?

File: app/components/registration/register.controller.js (hunk 7 above)
Comment (atreidas1) [inline app/components/registration/register.controller.js]: from server returns succes response  with needed data.

Comment (ghost) [inline app/components/registration/register.controller.js]: this code is incorrect, `var res` should be inside `.then` (I'm not sure that I understand what you expect to be in this var)

File: app/components/session.service.js [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    self.registration = registration;
+    self.logout = logout;
+
+    self.getUser();
+
+    function getUser() {
+      $http.get(API.BASE + API.AUTH + API.INFO)
+        .then(function (response) {
+          self.user.data = response.data.user;
+          $state.go("groups.list");
```
Comment (ghost) [inline app/components/session.service.js:35]: what will happen if I press `F5` and refresh browser page? I expect to be on the same page. Could you please check it?

File: app/components/session.service.js [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+      }
+    }
+
+    function getUser() {
+      $http.get(API.BASE + API.AUTH + API.INFO)
+        .then(function (response) {
+          self.user.data = response.data;
+          $state.go("groups.list");
+        })
+        .finally(checkUserView());
```
Comment (NikitaSedyx) [inline app/components/session.service.js]: it should be without `()`

File: app/components/logout.controller.js [hunk 12]
```diff
@@ -1,12 +1,17 @@
...
   angular
     .module("todo")
+    .controller("LogoutController", LogoutController);
 
-    .controller("LogoutController", LogoutController)
+  LogoutController.$inject = ["SessionService"];
 
-    LogoutController.$inject = ["$scope"]
+  function LogoutController(SessionService) {
+    var self = this;
```
Comment (NikitaSedyx) [inline app/components/logout.controller.js:9]: use `vm` instead of `self` for controllerAs

File: app/components/session.service.js (hunk 11 above)
Comment (ghost) [inline app/components/session.service.js]: :+1:

**Pull request merged**